addons/mail/static/src/discuss/core/common/message_patch.js

```diff
@@ -0,0 +1,23 @@
+/* @odoo-module */
+
+import { Message } from "@mail/core/common/message";
+import { patch } from "@web/core/utils/patch";
+
+patch(Message.prototype, {
+    /** @override */
+    enterEditMode() {
+        const body = new DOMParser().parseFromString(this.props.message.body, "text/html");
+        const mentionedChannelElements = body.querySelectorAll(".o_channel_redirect");
+        this.props.message.mentionedChannelPromises = Array.from(mentionedChannelElements).map(
+            async (el) => {
+                if (el.dataset.oeModel === "discuss.channel") {
+                    return this.store.Thread.getOrFetch({
+                        id: el.dataset.oeId,
+                        model: el.dataset.oeModel,
+                    });
+                }
+            }
+        );
```

Comment:
It would be cleaner to filter out oeModel before the map to avoid keeping undefineds longer than necessary.